nilearn/glm/tests/test_utils.py

```diff
@@ -31,12 +37,103 @@ def test_full_rank():
 
 
 def test_z_score():
-    p = np.random.RandomState(42).uniform(size=10)
-    assert_array_almost_equal(norm.sf(z_score(p)), p)
-    # check the numerical precision
-    for p in [1.e-250, 1 - 1.e-16]:
-        assert_array_almost_equal(z_score(p), norm.isf(p))
-    assert_array_almost_equal(z_score(np.float32(1.e-100)), norm.isf(1.e-300))
+    # ################# Check z-scores computed from t-values #################
+    # Randomly draw samples from the standard Student’s t distribution
+    tval = np.random.standard_t(10, size=10)
+    # Estimate the p-values using the Survival Function (SF)
+    pval = sps.t.sf(tval, 1e10)
```

Comment:
why 1e10 degrees of freedom?

Reply:
This number is the maximum accepted when we estimate the p-values in `nilearn/glm/contrasts.py` lines 257, 260, 261, 285, 286, 288, 289.